python/lsst/analysis/tools/tasks/sourceObjectTableAnalysis.py

```diff
@@ -20,19 +20,127 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 from __future__ import annotations
 
-__all__ = ("SourceObjectTableAnalysisConfig", "SourceObjectTableAnalysisTask")
+__all__ = (
+    "SourceObjectTableAnalysisConfig",
+    "SourceObjectTableAnalysisTask",
+    "ObjectEpochTableConfig",
+    "ObjectEpochTableTask",
+)
 
 import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
 import numpy as np
 import pandas as pd
 from astropy.table import vstack
-from lsst.pipe.base import NoWorkFound
+from lsst.drp.tasks.gbdesAstrometricFit import calculate_apparent_motion
 from lsst.pipe.base import connectionTypes as ct
 from smatch import Matcher
 
 from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
 
 
+class ObjectEpochTableConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "skymap"),
+    defaultTemplates={
+        # "inputName": "sourceTable_visit",
+        # "inputCoaddName": "deep",
+        # "associatedSourcesInputName": "isolated_star_sources",
+        # "outputName": "sourceObjectTable",
+    },
+):
+    refCat = ct.Input(
+        doc="Catalog of positions to use as reference.",
+        name="objectTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+        deferLoad=True,
+        deferGraphConstraint=True,
+    )
+
+    refCatEpochMap = ct.Input(
+        doc="Healsparse map of mean epoch of refCat.",
+        name="deepCoadd_epoch_map_mean",
+        storageClass="HealSparseMap",
+        dimensions=("skymap", "tract", "band"),
+        multiple=True,
+        deferLoad=True,
+    )
+
+    objectEpochs = ct.Output(
+        doc="Catalog of epochs for refCat objects.",
+        name="objectEpochTable",
```

Comment:
It would be nice if the `objectTables` had the epoch, but it's not possible to add that as part of this ticket. I don't see anything about not including `table` in dataset type names, unless you extend the point about catalogs to tables, but I'm fine calling this `object_epoch`.

Reply:
As I think about this point, we should define somewhere (if we don't already!) what each DR's epoch will be, and then report all of the `Object` astrometric values at that epoch, then we only need the "as measured" mjd_tai timestamps for each individual measurement and the mean of that as values to feed into that correction. I don't know where the plan for this would or should go.